### DISCUSSION TITLE
Which is the best way to work with two different parameters bound to the legend?

### DISCUSSION THREAD

**Discussion by eduardoraygoza-quidel** [vega/vega-lite#9057] [Q&A] [answered]
I have a line chart and I want to group the lines using different fields. I've achieved this using a `selection` parameter. [Example here](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQOJACckCCGgDaoWRrhoQAExoI4szO3lLiSQQ0PpT5y9YCMikExqzjaUF8cBkwjCDhBOChQpTYcKzktVG0TMwsE2U8lF3TrACYQAF0AX2KFPQMwiKiMr3DI6ICQTABPHCdlNl8YkAAzGgjjJJSyVTYGHCKlTDYyMkijTFVHEDLvX390SLILf2LC7KRMFFRQOwc4YdActzlPVHdstNvZAvQAYS9jI5PdEAAPNCPEAtIFlUCA1B5JSgh7ggFoADMMLB5QRqAALCi4WjIWJse59vCbhl7tDUq4Mm8QAAhL4-HQQoHYvLwyHk2Gs3FIllstBYkFoLlM1D4wVQolokkeHkU3JyakAQXpx0Z6OBsMRfKh2K13NQyPFepFAs12rFZpKJWm6lkEF6bFUCDVvUER0wFh0JgZBxASGGnW6WmtoCg9igDDdHqM30wDAQ2mlckKXn9RlG40m+yUyFUAGsjMJZIYlBYoGxTLIyE1IaB+oMjDgupYIJRAdM2h0AI4MJBuY5WUirbF1gaCTaBluUUEd9pGHt9qwDmhDtblwSOpr18fpsYTYcgOJIKB0WGhuSmWqnZRqDRVBq1Wz2FY4kDnF8ABlKxSAA)

If I select a group in the legend and then switch fields the chart goes blank. I understand that it's because the signal holds the values selected using the previous field and that I just need to select the groups in the new legend to show the lines, but I want to show all the lines (clear the values in the signal) when I change fields.

I thought about dividing the visualization in layers, create a param in each one to hold each legend values and set the whole visualization opacity to 0 if it's field is not selected, but the signals are bound to the same legend, so clicking on it adds or removes values from both signals. I tried setting `"toggle": "dimention == 'dimentionX'"` to just add values from their respective group, but `toggle` only works when working with multiple values, which is not quite what I need.

Is this the best way to handle this?
Would it be better to switch to full Vega?

Thanks!

**Comment by domoritz** [comment]
Does this work better if you instead change your data to long form (tidy from) and then switch the data using a selection over a field that switches between the two datasets?

**Reply by eduardoraygoza-quidel** [reply]
I hope I understood that correctly, following your advice here's what I did:

1. Added a `fold` transformation to change the data to long form
2. Added a `filter` on the `key` field to keep only the field I select using the combobox
3. changed the `color` property to use the `value` field

I get the exact same behavior, values from both fields can be added to the signal bound to the legend, am I missing something?

![image](https://github.com/vega/vega-lite/assets/131900118/f64db8b5-7591-4968-a780-3f121ab82f0e)

[Here's the updated chart](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQOJACckCCGgDaoWRrhoQAExoI4szO3lLiSQQ0PpT5y9YCMikExqzjaUF8cBkwjCDhBOChQpTYcKzktVG0TMwsE2U8lF3TrACYQAF0AX2KFPQMwiKiMr3DI6ICQTABPHCdlNl8YkAAzGgjjJJS7B0NCpUw2MjJIo0xVRxAy719-dEiyC39iiZMkTBRUUFHHYdActzlPVHdstKvZAvQAYS9jA6PdEAAPNDuQC1-mVQH9UHklEDbiDfmgAMyQ4HlWGoAAsiOhyLBYgx7l2MMuGRuENSrgyzxAACF3p8dKD-hi8jCwSSoUysfDGcy0OjAWh2fTUDi+eD8cjCR5OaTcnIKQBBGmHOkogFQuHc8EY9Uc1AIkXawW8tUa4XGkolSbqWQQXpsVQIZW2wTrFIS67vB7koow3qCA6YCw6fZKvZIYadbpaC2gfqCAOqIxu2QAAgAvKnkx9MAwEJQANZwIG7JTIVR5ozCWSGJQWKBsUyyMhNMExgbOow4LqWCCUP6TNodACODCQbkOVlIywxrcGHa7mB7QP77SMw9HVnHNEnKzrgjtTX6s-Qp0MKziSCgdChoDrfjo1iaKnUDvQ9RqD9s9iWmJAJ7QAAZSmKIA)

Here's also the chart divided in layers as I mentioned using this approach. This is the closest I've gotten to what I want to achieve, but clicking on the legend, even when the dimention selected is 2, it operates on signal 1. [Example](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQOJACckCCGgDaoWRrhoQAExoI4szO3lLiSQQ0PpT5y9YCMikExqzjaUF8cBkwjCDhBOChQpTYcKzktVG0TMwsE2U8lF3TrACYQAF0AX2LC7KRMFFRQOwc4JN1U1wzPVHdstLc5AvQAYS9jSuqmgA80DpAATwnihVBx1DylGfa5hbQAZhXZ+ZBFgBYdtb3FsWP3MvXm3Lk25Zvu2V6QACFB4Z0N9uO868WHqs-qctr9-mgjtM0MDvucoUsrnsck82ttHhkXgBBD5VL77CbHTbgpaE4lo1ZEkGoSEU4lw2klEpKTDqWQQABmbFUCDxnME-mS6I8gy6GKK13ZgkqmAsOhMn3KICQjWUbF8mC0TNA7JoghlqiMyIyAAIALym41DTAMBCUADWcBmZSUFigbFMsjIAXxNRAOoiAtV6oglHGzKmOCcIAAjgwkG4qlZSCA5vDtTQA0YcGrLCGZuHI0ZY-GrImaMnSkopVM4AbkqBkKo7UZhLJDEoVOpNHj9OZDaKPOFItFrF4h1FQr7MBGo9n1V5-fyVUbhYrMGwyGRIkYWY4U0ofH4W+QLP5nSAWfGOVyefW-br9YbKjb7Y6zRaAOQru4flOK13ur4Xq+m6ghct6i6Bt+mT7iAcRIFAdCrKAbp+HQo6+p2Gj9i0g4RBOo62PYe4nCAdQkQADKU1yNs26Ctu2yhqBojR6AYOG3M844jnIY74dE3rToW6BzpYC4ZkucrQQUa4bluUa7oYqaHoGkRkKef7Mqy17cryD61k+1q2g6UzvsaX4Dj0v7ngBHrAShbBgXW6aZs4lnPLB8GIdO3qoaYGTelht5Cj03EBUR9S7GRxFOFRpQlEAA)

Any ideas are appreciated!

**Comment by eduardoraygoza-quidel** [comment, marked as answer]
I've managed to solve it with help from another forum, here's the solution:

1. Compile Vega-Lite to Vega
2. To any signal that listens to `view`-`click` event add an additional `dimention` event to the listener, and in that case return `null`

[Solved example](https://vega.github.io/editor/#/url/vega/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykSArJQBWENgDsQAGhBMkUANZkATmwbiAJmhAB3GHTjSQOJBo01xZNAJk6aG+mgBMABhcz4NMlmduZETABPABtDdCg4YOCjDSRMFFQAbVBxJAQwkAhIuChMdnEAfQC2ZUMAXykUtIyxBmUIgpcjYiRghjgINGSQc3TxPIkARjRBmV64fvynbQBhGLiE7oAPEZlAkYrQFdQnNY3KkG2AZj3UQc3DtAAWU-OD7ZsQdbOygF0L8cmh5zGaPoHxNN0AAhebxLpbVZPZwXba7aE7WFoE4IpxI1A3VHox7PNHvA6fAHDVAowlTbQAQTBi0hZ1OR3R8OeDPuyPp6MxzOx7Pe+Kq6W0sXijSMtXqNTU4pFMkwyiQ4ggADMSggISAgjgMorgnFMBMjIqaJENJ0kj0FiBXjIkKbEsY2BZMJ0+erAprtMrlAgGDqjHAljhlIK4gwEIkADr2x0QSOvAAEADIE3GhaGI1H+jHLemlrGjDbtDgHZnKCsLhqtSqfSgZP7A8HMGnI0Xo7HE8nU2Hm8WnbH04E89bTRmnZR1uW3ZWvdW-QGg+hO4kyRJXvnhyo1DgQG83v4vKlgrb+Rl1HRmq12mhgBUQBI1XBSJntAg1FkX6QjAwcEKMjQIAAVL9QgACg3L9gIASgguMAH44zAnBILjVA41PTBtxdVIBXQLJQlyfICgQgpQjICYtBkFo2jCcQfWCGQ7ySUAHwmJ01TFCJtGII0dCMCttCgYIaEUIxkGUBQsIyIiSLIopAgQJg2EPbcDnYjIuLgHiZUnfjBOEmRRPE6ptCk8gZJ1JhIk6C5VM47jeO08JdIUESkDEiTjNUL9iNMzQChY5QjSsq0QC-H9tD-AAROBDXEOANGAztKEo9poLgxLkrgZC4wMBBINYPUEAgRIXFefLECKkrKHSi9DBkT0ONQWV2guZin0YzJJQakB1M0113UcoTnN3ELvziDIAEJWswMqEDjAAfOa43GztYLjGioiy3CcgBQjPJwbzSM0A0Sgapq4Bax9WNQUAIH3VpBT+Fj8mUkawvQdbojqk6wjOncDncnDsnwu8ZFCsbOOCABlIGAQAJQ6RTSGAyMtuBwpilKSMpDjSNTwkLG4zO7Gzog5Tjwe-5noomqKae74ZCYCwtGukALBwBh0MBvD0PonAAVtHpHq+cRhl+SmJGmd4LgBzIYYIxscFCT9Rr1bRUZ2kzDo0JaAF4dbW2jVuAQ1jXwON1floC4AKE3ghNbGMrNxILYkXbNwOsj3iyj6yZAGWXcKBXQhto07c6amqLVPj0AAUQNUPmZABCMOloyue2+W2DIMglYjy9UEVVosnoyR2qm01QHSZRSLYzq1LsrT+pAATBpT0GVYyM7ffLq9Mju6IFyFgEXrB1X3to47xTQX7MLT2XuYIl9zEVdYS-vS6K77sgDzVuXXaD8p27ekAl5oFfkfnjPXYxuACYDgpMCznPMrgj7Nr3wOrex+-H+z0JVrfihIm5sP4Pyfv-OCP8rbe1oqTHcwUDJHj9nPEw9Bk6N0kntA0qhVQs0LhETm5N0D4LgHqDQBRUEwAKMgCw1IGxIGlEnPaTBnh2mTlLWs4goBsA0GEUAo8+G6HsI4PBCde7J3QHYBwMBtw3k8N4QhIBbbM1ABIjAcAvA+DKNom8iC1Qyz0Rg7QglYqiiCErM0JjDDBTEMoRRyiGxNhHBAUseYbyOjgHKfCH5GrKHaHVHBvchQJCUfIUhcUKFxCoTQyQN4JjcN4b3ARvcAiqAUII6ArQMjcOCCUeOxoPKbhemwEwUA6CsNAHqAI2hxqhEsPQBKCwL73xvpGKC81FrEChh-f8HRMDNNAa06QKYQwIFJnnMIdxQAZTQC4YavDoABT5s9Fmt1t73XQN2aMpYUKRjjAAajjMBT0yB+mLi2SWXMlpsaRjadBQ5kYADccYWwlkCLskAByjknLiI0xsXZnFjljDckAdyvlPPgntD5XzgJ-gAGqtHsH8pxCFYypRGf89MqLLRZVuSAfZ5ymGbjRS9bYN0oBZO0CsOqYj0CvN7OGUsL1njkspegVeSjaXOPDIy8cYxooWDiik-u4UIAIsEvFQl9KXFXNeNBJM2UIAADELAGGAgS0Z6ZpWuMtPK5M8LEWSs1Rc0cA5dXtkVSq8QaqNWYpNS4s1crZHaN5P4CloQkEy2pX1DIVjXIxDYDEoJFoFwLEYQ4ulPYYyMrLDIOUlgwjFRUiKyRwiZHDWtadPxcAZAAC9PFsDQIXQ851-pzw5dHEAfqgxjEDUgWhLNgn0PDVy6VPKxwvXjTXdqayd7oHkVoqQJUZCZp+tmvNBbp7ZtTthZuik8lGPQCUcwO9a1BsbSG80womg0oKegdBHU7FTuanG+U3bm5jTICUccwUkBLA6Gxd1GRvUlCNP0bQClsCBqMCoewkMn3aA5b+5mRN1SDRmGod9qyU3Nw0cEYCUj6AECuC4OBa762l2LcXKtSALJKQLkXHNIBXI0ASFhojyAlgxyWHqKD7gT4WGo7Rzm9G8iKFNORvNTN-SzOsgB9lRhX0sWMdFHmTC-38cuDIYDx6iNsYUBB9Qije0bNgzQeDA7MDIdQy9DQdaG2cZw3hjjhHrQBTI6Zk+d6mPCdQPRhAjGaO2dY4NEzJauOaB43ZvjbKpO3gCrZ2QbAv2qmkwFZmhm8iYAsYC715lIhKraBAGRvj-FGciAAeVIMoHUW5UtyfA5B5TMGIjqYQ2m7TaGQC5u49sOZKlJMcqE1Bqtomf3haLZZqLMXtUcvi8ELLnjcuyZlIVpTwr1kDzUxpjRCjKsvRq55urw1pKaCQaktg6T+LzprZkOSClgj-gcpkWUm3apgei5gopnCEmCP65vGWmszK4cskYDxXi8g+NA8klmJT5DlKjn07QwFakTDIA04JAyF7X0fpjEA7SFpHIDlivabYFW1bYIqYCyPIzYqtBi0MSUarQQAHz61Q+M7qNMXgHBmXZygRwBA7hvBAfbikHtzye75Vn8l2dvf6J4+Qn2x1pZ+6AP7ZSgiA+qegEHdTwcwD+UgKHV90aw9vvD6CiPscfxR8SnF6OluY519D8QeuvzAoJwgInVFSfk6gueSONPpnU5cAzpnHCQD+UCr3R7e0Pa+W9w+mQ72hc0C++O164NRFRF7nT9U8aIAmFKO+nRLqyhAA)

Kudos to [Marco Liberati](https://discuss.elastic.co/u/Marco_Liberati)
